app/frontend_management.py

```diff
@@ -159,16 +168,19 @@ def init_frontend_unsafe(cls, version_string: str, provider: Optional[FrontEndPr
             Path(cls.CUSTOM_FRONTENDS_ROOT) / provider.folder_name / semantic_version
         )
         if not os.path.exists(web_root):
+            tmp_path = web_root + ".tmp"
```

Comment:
Please clearly document that why the ".tmp" move is necessary. i.e. put a comment here describing the considerations stated in 

> Also moves the download to a folder with a .tmp suffix until everything's extracted as otherwise you can CTRL+C the comfy window while it's downloading and get stuck in a state where comfyui thinks its has the frontend downloaded but it doesn't actually.